torchao/quantization/quant_primitives.py

```diff
@@ -389,13 +389,10 @@ def _quantize_affine_no_dtype_cast(
         assert (
             zero_point is None
         ), "zero_point should be None when zero_point_domain is NONE"
-        quant = torch.clamp(torch.round(input * (1.0 / scale)), quant_min, quant_max)
-    elif zero_point_domain is None:
-        # This case handles quantization for float8 we expect no zero point and no zero point domain
-        assert (
-            zero_point is None
-        ), "zero_point should be None when zero_point_domain is None"
-        quant = torch.clamp(input * scale.reciprocal(), quant_min, quant_max)
+        if _is_float8_type(input.dtype):
```

Comment:
This check is incorrect because it'd work in the dequantize case, and not the quantize case (this one). So I'll fix it.

Is there some way to ascertain if we're trying to quantize to float8 here? Sorry, this path requires an Nvidia GPU of a higher Compute Capability than the one I have locally. Thanks!

cc @drisspg

Reply:
It seems to me like the main behavioral change we want here when there is no zero point domain is to know whether we should round or not round the scale values before clamping. Is that correct?

Reply:
Yes, that's right. Thanks!

Reply:
@drisspg, but this behavior differs for float8 and non-float8 quantization, so I'm wondering how to differentiate both cases, since the output dtype is not exposed to this function (viz. `_quantize_affine_no_dtype_cast`). Thanks!

Would it be okay to match `quant_min` & `quant_max` with that of FP8 types? I feel like practically, doing so would not lead to any side-effects. Thanks!

Reply:
we really should be separating out these ops so it's easier to maintain I think, especially for fp8, we'd want separate quant primitive ops, cc @jainapurva 

for now I think whatever passes test would be OK. the target dtype is only available out side of this function: https://github.com/pytorch/ao/blob/de5c6e12f3f8cf3ce3d682880873034baf3cb4ad/torchao/quantization/quant_primitives.py#L338, so there is no way to distinguish it from here, I think you'll need to pass around a flag from the callsite of this function instead